layout/shims.proj

```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+
+  <UsingTask TaskName="PrereleaseResolveNuGetPackageAssets" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <ItemGroup>
+    <!-- classic assembly refs -->
+    <ShimAssembly Include="mscorlib" />
+    <ShimAssembly Include="System" />
+    <ShimAssembly Include="System.Core" />
+    <ShimAssembly Include="System.Runtime.Serialization" />
+    <ShimAssembly Include="System.Xml" />
+    <ShimAssembly Include="System.Xml.Linq" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <RestoredRefPath>$(IntermediateOutputPath)CompatShims/ref/</RestoredRefPath>
+    <GenFacadesOutputPath>$(BinDir)CompatShims/facades/</GenFacadesOutputPath>
+    <NetCoreAppRefPath>$(LayoutBasePath)ref/netcoreapp1.1/</NetCoreAppRefPath>
+    <ProjectJson>$(MSBuildThisFileDirectory)project.json</ProjectJson>
+    <ProjectLockJson>$(MSBuildThisFileDirectory)project.lock.json</ProjectLockJson>
+  </PropertyGroup>
+
+  <Target Name="RestoreAndPublishForShims"
+          Inputs="$(ProjectJson)"
+          Outputs="$(ProjectLockJson);$(RestoredRefPath)">
+
+    <Exec Command="$(DnuRestoreCommand) $(ProjectJson)"
```

Comment:
I think of this as akin to the individual project restore. This isn't a traditional project so I'm not sure if it will plug in nicely but I will take a look.